### PR TITLE
Added io.github.igaldino.CoyoteTM

### PR DESCRIPTION
**Pull request by igaldino**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.CoyoteTM.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
+{
+    "app-id" : "io.github.CoyoteTM",
+    "runtime" : "org.freedesktop.Platform",
+    "runtime-version" : "19.08",
```
Comment (hfiguiere) [inline io.github.CoyoteTM.json:4]: Please use runtime `20.08`

File: io.github.CoyoteTM.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    "sdk" : "org.freedesktop.Sdk",
+    "command" : "coyotetm",
+    "tags" : [
+        "nightly"
+    ],
+    "desktop-file-name-prefix" : "(Nightly) ",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (hfiguiere) [inline io.github.CoyoteTM.json:14]: I don't think fltk supports wayland.

File: io.github.CoyoteTM.json [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        "--device=dri",
+        "--share=network",
+        "--talk-name=org.gtk.vfs",
```
Comment (hfiguiere) [inline io.github.CoyoteTM.json:17]: this is not needed (should be removed). It was never used.

File: io.github.CoyoteTM.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        "--talk-name=org.gtk.vfs.*",
+        "--filesystem=host",
```
Comment (hfiguiere) [inline io.github.CoyoteTM.json:19]: Do you really to give access to the whole host.

File: io.github.CoyoteTM.json [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            "buildsystem" : "meson",
+            "config-opts" : [
+                "--libdir=lib"
+            ],
+            "builddir" : true,
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/igaldino/coyotetm.git",
+                    "branch" : "coyotetm-1-0"
```
Comment (hfiguiere) [inline io.github.CoyoteTM.json:88]: You have to specify a commit.

File: io.github.CoyoteTM.json [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            "subdir" : "build_unix",
+            "build-commands" : [
+                "../dist/configure --prefix=/app --enable-compat185 --enable-dbm --disable-static --enable-cxx",
+                "make",
+                "make install"
+            ],
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "http://download.oracle.com/berkeley-db/db-6.2.23.tar.gz",
```
Comment (hfiguiere) [inline io.github.CoyoteTM.json:72]: Please use https to download this.

File: io.github.CoyoteTM.json [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            "buildsystem" : "autotools",
+            "config-opts" : [
+                "--enable-xdbe",
+                "--enable-xft"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/fltk/fltk.git",
+                    "branch" : "branch-1.3"
```
Comment (hfiguiere) [inline io.github.CoyoteTM.json:56]: Please specify a tag or a commit.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build io.github.CoyoteTM

File: io.github.CoyoteTM.json (hunk 3 above)
Comment (hfiguiere) [reply io.github.CoyoteTM.json:17]: Also are you really using gvfs since it's a fltk app?

**Comment by hfiguiere** [comment]
Buld fail because it can't find the appstream file or it doesn't have the right app-id.

Looking at this, the id is wrong:
https://github.com/igaldino/coyotetm/blob/main/data/appdata/io.github.CoyoteTM.appdata.xml

(there is no `.desktop` in the id)

File: io.github.CoyoteTM.json [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options" : {
+        "cflags" : "-O2 -g",
+        "cxxflags" : "-O2 -g",
+        "env" : {
+            "V" : "1"
+        }
+    },
```
Comment (barthalion) [inline io.github.CoyoteTM.json:31]:
```suggestion
```

File: io.github.CoyoteTM.json [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network",
+        "--talk-name=org.gtk.vfs",
+        "--talk-name=org.gtk.vfs.*",
+        "--filesystem=host",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (barthalion) [inline io.github.CoyoteTM.json:23]:
```suggestion
```

It should work by default with `20.08`.

File: io.github.CoyoteTM.json (hunk 4 above)
Comment (barthalion) [reply io.github.CoyoteTM.json:19]: README says `~/coyotetm` so maybe that will suffice?

File: io.github.CoyoteTM.json [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+                    "type" : "git",
+                    "url" : "https://github.com/fltk/fltk.git",
+                    "branch" : "branch-1.3"
+                }
+            ]
+        },
+        {
+            "name" : "libdb",
+            "buildsystem" : "simple",
+            "subdir" : "build_unix",
```
Comment (barthalion) [inline io.github.CoyoteTM.json:63]: Hm, any reason not to specify `dist` here and keep buildsystem set to `autotools`?

File: io.github.CoyoteTM.json (hunk 3 above)
Comment (igaldino) [reply io.github.CoyoteTM.json]: fixed

File: io.github.CoyoteTM.json (hunk 4 above)
Comment (igaldino) [reply io.github.CoyoteTM.json]: done

File: io.github.CoyoteTM.json (hunk 2 above)
Comment (igaldino) [reply io.github.CoyoteTM.json]: fixed

File: io.github.CoyoteTM.json (hunk 1 above)
Comment (igaldino) [reply io.github.CoyoteTM.json]: done

File: io.github.CoyoteTM.json (hunk 5 above)
Comment (igaldino) [reply io.github.CoyoteTM.json]: changed

File: io.github.CoyoteTM.json (hunk 6 above)
Comment (igaldino) [reply io.github.CoyoteTM.json]: done

File: io.github.CoyoteTM.json (hunk 7 above)
Comment (igaldino) [reply io.github.CoyoteTM.json:39]: changed

File: io.github.CoyoteTM.json (hunk 8 above)
Comment (igaldino) [reply io.github.CoyoteTM.json]: done

File: io.github.CoyoteTM.json (hunk 9 above)
Comment (igaldino) [reply io.github.CoyoteTM.json]: done

File: io.github.CoyoteTM.json (hunk 10 above)
Comment (igaldino) [reply io.github.CoyoteTM.json:46]: yep, the configure script must be run outside of dist directory.

**Comment by igaldino** [comment]
> Buld fail because it can't find the appstream file or it doesn't have the right app-id.
> 
> Looking at this, the id is wrong:
> https://github.com/igaldino/coyotetm/blob/main/data/appdata/io.github.CoyoteTM.appdata.xml
> 
> (there is no `.desktop` in the id)

humm, is this an hard requirement? I see that other projects also have .desktop in the id.

**Comment by igaldino** [comment]
bot, build io.github.CoyoteTM

**Comment by igaldino** [comment]
bot, build

**Comment by igaldino** [comment]
bot, build io.github.igaldino.CoyoteTM

**Comment by igaldino** [comment]
> (there is no `.desktop` in the id)

Fixed.

**Comment by hfiguiere** [comment]
bot, build io.github.igaldino.CoyoteTM

**Comment by igaldino** [comment]
bot, build io.github.igaldino.CoyoteTM

**Comment by igaldino** [comment]
So, I've done a few changes upstream. I hope this fixes all the remaining issues. Thx folks.

**Comment by hfiguiere** [comment]
It wants a 128x128 icon. Or a SVG (scalable)

**Comment by igaldino** [comment]
bot, build io.github.igaldino.CoyoteTM

**Comment by igaldino** [comment]
Hello folks.

Do I need to anything else here or are we good?

Thanks @barthalion and @hfiguiere for the reviews.

**Comment by barthalion** [comment]
Err, it was supposed to be merged. Let's try again.

/merge

**Pull request closed without merging**